### DISCUSSION TITLE
Using PySR on a trained feedforward neural network

### DISCUSSION THREAD

**Discussion by gurtajbir** [MilesCranmer/PySR#471] [Q&A] [answered]
Hi. I have a single-layer feedforward neural network that was fitted on a dataset (created using known functions). I want to recover the generating functions from the outputs of the trained neural network (and not the output from the dataset). Is it possible to do so? If yes, are there any considerations/guidelines that might aid this search as opposed to if we were to use PySR directly on the dataset? Thanks!

**Comment by MilesCranmer** [comment, marked as answer]
Hey @gurtajbir,

Yes, you would do this by recording the inputs and outputs from the network. This is what is done on https://github.com/MilesCranmer/symbolic_deep_learning (see end of example notebook notebook)

Here is a [slide](https://speakerdeck.com/milescranmer/symbolic-distillation-of-neural-networks?slide=38):
<img width="1390" alt="Screenshot 2023-11-19 at 16 24 24" src="https://github.com/MilesCranmer/PySR/assets/7593028/ac0e83a1-874c-4001-8abf-c2ef39172bc5">

**Reply by gurtajbir** [reply]
Thank you @MilesCranmer for your response. I have a follow-up question: Does it matter if the network whose output will be used with PySR is overfitted on the data it is trained on? Not sure about this but I can't convince myself of how a more generalizable network would be better when talking specifically about the inputs and outputs of the training set.

**Reply by MilesCranmer** [reply]
> Does it matter if the network whose output will be used with PySR is overfitted on the data it is trained on

Good question. So even if you are evaluating the network on the training set, you still would want to avoid overfitting the network (but it depends on problem I think). Just think about adding noise to samples of a sinusoid — you wouldn’t necessarily want the network to fit the points exactly, otherwise you’d be fitting PySR to the noise as well
